apps/vmq_commons/include/vmq_types.hrl

```diff
@@ -8,19 +9,7 @@
 -define(true, 1).
 -define(false, 0).
 
--type peer()                :: {inet:ip_address(), inet:port_number()}.
--type username()            :: binary() | undefined.
--type password()            :: binary() | undefined.
--type client_id()           :: binary().
--type mountpoint()          :: string().
--type subscriber_id()       :: {mountpoint(), client_id()}.
--type reg_view()            :: atom().
--type topic()               :: [binary()] | binary().
--type qos()                 :: 0 | 1 | 2.
--type routing_key()         :: [binary()].
--type payload()             :: binary().
--type flag()                :: ?true | ?false | boolean() | empty. % empty for test purposes
--type msg_ref()             :: binary().
+-type fflag()               :: flag() | ?true | ?false | empty. % empty for test purposes
```

Comment:
yeah you're right.
`flag()` is kind of the high level type that is used at multiple places within `vmq_server`. The goal is, that we can ensure that all `vmq_server` sees is a `boolean()` and not `0|1`. However the parser deals with flags containing `0|1` as well as the conversion from `true|false -> 1|0` the `empty` type is for testing purposes.